src/main/java/com/epam/test_generator/services/StepSuggestionService.java

```diff
@@ -73,11 +74,15 @@ public StepSuggestionDTO getStepsSuggestion(long stepSuggestionId) {
         return stepSuggestionTransformer.toDto(stepSuggestion);
     }
 
-    public List<StepSuggestionDTO> getStepsSuggestionsByType(long typeId) {
-
-        return stepSuggestionTransformer.toDtoList(stepSuggestionDAO.findAll().stream()
-                .filter(stepSuggestion -> ((long)stepSuggestion.getType())== (typeId))
-                .collect(Collectors.toList()));
+    public List<StepSuggestionDTO> getStepsSuggestionsByType(StepType stepType) {
+        return stepSuggestionTransformer.toDtoList(
```

Comment:
this piece of code is hard to read, consider following:

        return stepSuggestionTransformer.toDtoList(
                stepSuggestionDAO.findAll().stream()
                        .filter(s -> stepSuggestion.getType() == stepType))
                        .collect(Collectors.toList()));

Reply:
Fixed